src/utils/validators_enhanced.py

```
@@ -0,0 +1,453 @@
+        self.warnings.append(warning)
+
+
+def _with_timeout(func, timeout: float = VALIDATION_TIMEOUT):
+    """Decorator to add timeout protection to validation functions."""
+    def wrapper(*args, **kwargs):
+        start_time = time.time()
+        result = func(*args, **kwargs)
+        elapsed = time.time() - start_time
+        
+        if elapsed > timeout:
+            raise ValidationError(f"Validation timeout exceeded: {elapsed:.3f}s > {timeout}s")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)